pettingzoo/utils/conversions.py

```diff
@@ -236,3 +244,43 @@ def close(self):
 
     def __str__(self):
         return str(self.env)
+
+
+class turn_based_to_parallel_wrapper(to_parallel_wrapper):
+    def reset(self):
+        self.aec_env.reset()
+        self.agents = self.aec_env.agents[:]
+        observations = {agent: self.aec_env.observe(agent) for agent in self.aec_env.agents if not self.aec_env.dones[agent]}
+        return observations
+
+    def step(self, actions):
+        if actions == {}:
+            rewards = defaultdict(int)
+            dones = dict(**self.aec_env.dones)
+            infos = dict(**self.aec_env.infos)
+            for agent in self.aec_env.agents:
+                infos[agent]["active_agent"] = self.aec_env.agent_selection
+            observations = {agent: self.aec_env.observe(agent) for agent in self.aec_env.agents}
+            return observations, rewards, dones, infos
+        self.aec_env.step(actions[self.aec_env.agent_selection])
+        rewards = dict(**self.aec_env.rewards)
+        dones = dict(**self.aec_env.dones)
+        infos = dict(**self.aec_env.infos)
+        observations = {agent: self.aec_env.observe(agent) for agent in self.aec_env.agents}
+
+        while self.aec_env.agents:
+            obs, rew, done, info = self.aec_env.last()
+            agent = self.aec_env.agent_selection
+            observations[agent] = obs
+            rewards[agent] = rew
```

Comment:
Very subtle, but this is actually incorrect. This will set the cumulative reward, but the agent already got its reward from other player's turns. The reward scheme needs to be consistent.